modes/company/evil-collection-company.el

```diff
@@ -62,9 +62,13 @@ be set through custom or before evil-collection loads."
 (defun evil-collection-company-supported-p (command &rest _)
   "Return non-nil if `evil-state' is in supported states."
   (cond
-    ((eq command 'prefix)
-     (memq evil-state evil-collection-company-supported-states))
-    (t t)))
+   ((eq command 'prefix)
+    (memq evil-state evil-collection-company-supported-states))
+   (t t)))
+
+(defun evil-collection-company-popup ()
+  "When `evil-mode' is active, make `company-mode' not show popup if not in supported state."
+  (advice-add 'company-call-backend :before-while 'evil-collection-company-supported-p))
```

Comment:
It might be better to modify evil-collection-company-supported-p to check if evil-mode is on first (e.g. (and (bound-and-true-p evil-mode) (memq evil-state evil-collection-company-supported-states)

Reply:
This is a good solution, but it causes the same problem. It changes the company-call-backend when I enter `evil-mode` and doesn't change it back after `evil-mode` gets deactivated, so it still sticks. I will look at this with a fresh pair of eyes in the morning.

Reply:
Why does it need to change? The function itself should be handling the logic, no?

Reply:
Yes, the function should be handling the logic. What I meant is

```emacs-lisp
(and (bound-and-true-p evil-mode) (memq evil-state evil-collection-company-supported-states))
```
won't work as it is. There should be an `or` statement to ensure that if `evil-mode` is not active then this function doesn't return `nil`. Adding to what you suggested, something along the line:
```emacs-lisp
(or (and (bound-and-true-p evil-mode) (memq evil-state evil-collection-company-supported-states))
 (not (bound-and-true-p evil-mode)))
```
looks to work. I am going to try this out and see what happens.

Reply:
I just pushed an update.

Reply:
You can just add an additional clause checking explicitly if evil-mode is active and return true if it's not active instead of making the current clause more complex.

Reply:
That could work too. I'm new to Emacs so thanks for the info, I appreciate it. I'll check if it works and push an update.